docs/articles/documentation/using-testcafe/common-concepts/connect-to-the-testcafe-server-over-https.md

```diff
@@ -0,0 +1,60 @@
+---
+layout: docs
+title: Connect to the TestCafe Server over HTTPS
+permalink: /documentation/using-testcafe/common-concepts/connect-to-the-testcafe-server-over-https.html
+---
+# Connect to TestCafe Server over HTTPS
+
+TestCafe is a proxy-based testing tool. Browser requests are sent via the TestCafe proxy server to the tested website. All requests between the browser and the TestCafe server are sent over the HTTP protocol.
+
+![Connection Protocols](../../../images/proxy-connection-protocols.svg)
+
+Some browser features (like
+[Service Workers](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API),
+[Geolocation API](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API),
+[ApplePaySession](https://developer.apple.com/documentation/apple_pay_on_the_web/applepaysession), or
+[SubtleCrypto](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto))
+require secure origin. This means that the website should be served over the HTTPS protocol. If you run tests against pages with this kind of browser API in a regular way, these tests fail with JavaScript errors.
```

Comment:
"This means that the website should be served over the HTTPS protocol." Served over?

Reply:
"in a regular way"?

Reply:
"these tests fail with JavaScript errors" Tests fail with errors?

Reply:
1. `served over HTTPS`

    I've picked this up from our developers. I've googled a bit and found this expression in some articles (https://www.itnews.com.au/news/firefox-blocks-insecure-content-served-on-https-352592) and in Google documentation describing error messages (https://support.google.com/blogger/answer/6286127?hl=en).

    Another popular alternative is `served on HTTPS`. If you feel like it's better - I can use it instead.

2. `in a regular way`

    You're right, I should be more specific

3. `Tests fail with errors`

    Hm... I thought this is a correct phrase, but probably it is a calque from Russian.

**Version 2**

Some browser features (like [Service Workers](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API), [Geolocation API](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API), [ApplePaySession](https://developer.apple.com/documentation/apple_pay_on_the_web/applepaysession), or [SubtleCrypto](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto)) require secure origin. This means that the website should be served over the HTTPS protocol. If you run tests against pages with this kind of browser API through HTTP, these tests fail because of JavaScript errors.

Reply:
Some browser features (like Service Workers, Geolocation API, ApplePaySession, or SubtleCrypto) require a secure origin. This means that the website should use the HTTPS protocol. Pages that are tested with this kind of browser API through HTTP fail because of JavaScript errors.

Reply:
> Pages that are tested with this kind of browser API through HTTP fail because of JavaScript errors.

Sorry, my bad, my sentence was ambiguous. *Pages with this kind of browser API* - not *tested with this kind of browser API*. And *tests fail because of JavaScript errors* - not *pages fail...*

Also, `test through HTTPS` seems not strict enough.

So, we need one more version. How about this (got rid of *pages*)?

Some browser features (like Service Workers, Geolocation API, ApplePaySession, or SubtleCrypto) require a secure origin. This means that the website should use the HTTPS protocol. If TestCafe proxies such a website through HTTP, tests fail because of JavaScript errors.